### PR TITLE
Fixes #533 -- include potential alternatives in the mismatched report

### LINKED ISSUE #533: Add support to mismatched script for recognizing potential alternates (1/22)

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by alex** [comment]
Right now it does the "nearby date" for NOD. NEed to add form9 support and then do the alternate label issue.

**Comment by paultag** [comment]
Should we add this to the test JSON, too? It might be nice to have a `joe-snuffy` with a NOD-mislabel

**Comment by alex** [comment]
@paultag is that useful? the fakes are only used in the web interface ATM.

**Comment by paultag** [comment]
@alex not currently, but it might be good if we're thinking we would want to build out a UI that suggests a document might just be mislabeled

**Comment by alex** [comment]
Code is basically complete at this point. Needs confirmation on a few constants before it's ready so I would say review can wait.

**Comment by gmorgana-va** [comment]
Date and label parameters have been confirmed:  https://github.com/department-of-veterans-affairs/appeals-pm/issues/98

**Comment by alex** [comment]
:+1:

On Mon, Jan 25, 2016 at 11:08 AM, Giuseppe Morgana <notifications@github.com
> wrote:

> Date and label parameters have been confirmed:
> department-of-veterans-affairs/appeals-pm#98
> <https://github.com/department-of-veterans-affairs/appeals-pm/issues/98>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/department-of-veterans-affairs/caseflow/pull/544#issuecomment-174556714>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

File: lib/sources/caseflow/reports.rb [hunk 1]
```diff
@@ -145,6 +196,8 @@ def spreadsheet_cells(vacols_case)
           Caseflow::Reports.hearing_pending(vacols_case),
           vacols_case.bfcorlid,
           Caseflow::Reports.bool_cell(vacols_case.merged?),
+          Caseflow::Reports.potential_date_alternatives(vacols_case).join(", "),
+          Caseflow::Reports.potential_label_alternatives(vacols_case).join(", "),
```
Comment (paultag) [inline lib/sources/caseflow/reports.rb]: I don't know I want to fix this, but it's intentional we're returning alternatives even if we have a hit on the exact document type, right?

We're sure this won't cause anyone to get confused about "Why are there things for NOD in `POTENTIAL LABEL ALTERNATIVES` if there's no mismatch for the NOD", right? (If not, it's likely setting expectations and not short circuiting)

Comment (alex) [inline lib/sources/caseflow/reports.rb]: I don't think that's the case, both `potential_date_alternatives` and `potential_label_alternatives` check if that document is in `mismatched_dates(c)` before doing anything.

**Pull request merged**